platform/ios/src/MGLMapView.mm

```diff
@@ -2594,6 +2645,12 @@ - (void)removeAnnotations:(NS_ARRAY_OF(id <MGLAnnotation>) *)annotations
         {
             continue;
         }
+       
+        MGLAnnotationContext &annotationContext = _annotationContextsByAnnotationTag[annotationTag];
```

Comment:
Always call `count()` on the `map` before indexing into it, or call `at()` instead of indexing into it directly. As written, if `_annotationContextsByAnnotationTag` has no entry for `annotationTag`, this code [will create one](http://www.cplusplus.com/reference/map/map/operator[]/) using the default constructor, throwing off logic everywhere else. ([`at()`](http://www.cplusplus.com/reference/map/map/at/) throws an exception if no entry is found.)